x-pack/plugins/enterprise_search/server/routes/enterprise_search/api_keys.ts

```diff
@@ -54,9 +54,19 @@ export function registerApiKeysRoutes(
       const { client } = (await context.core).elasticsearch;
       const user = security.authc.getCurrentUser(request);
       if (user) {
-        const apiKeys = await client.asCurrentUser.security.getApiKey({ username: user.username });
-        const validKeys = apiKeys.api_keys.filter(({ invalidated }) => !invalidated);
-        return response.ok({ body: { api_keys: validKeys } });
+        try {
+          const apiKeys = await client.asCurrentUser.security.getApiKey({
+            username: user.username,
+          });
+          const validKeys = apiKeys.api_keys.filter(({ invalidated }) => !invalidated);
+          return response.ok({ body: { api_keys: validKeys } });
+        } catch {
+          // Ideally we check the error response here for unauthorized user
+          // Unfortunately the error response is not structured enough for us to filter those
+          // Always returning an empty array should also be fine, and deals with transient errors
+
+          return response.ok({ body: { api_keys: [] } });
```

Comment:
This could be a little misleading if they don't have permissions and are say, using API keys elsewhere. But I think it's an OK workaround for now. Thanks for fixing!

Reply:
Yeah it's not ideal. Thankfully security is working on a few utilities we can lean on going forward.